cli/flox-rust-sdk/src/models/manifest.rs

```diff
@@ -500,12 +553,101 @@ pub struct ManifestInstall(
     BTreeMap<String, ManifestPackageDescriptor>,
 );
 
+#[derive(Debug, Clone, Serialize, Deserialize, PartialEq, Eq, Hash)]
+#[cfg_attr(test, derive(proptest_derive::Arbitrary))]
+// todo: this can make the error messages less clear and might call for a custom (de)serialize impl
+#[serde(untagged)]
+pub enum ManifestPackageDescriptor {
+    Catalog(ManifestPackageDescriptorCatalog),
+    FlakeRef(ManifestPackageDescriptorFlake),      // todo
+    StorePath(ManifestPackageDescriptorStorePath), // todo
```

Comment:
blocking: per https://flox-dev.slack.com/archives/C07525NJ73L/p1718388628400479?thread_ts=1718379366.989469&cid=C07525NJ73L
```suggestion
    // TODO StorePath(ManifestPackageDescriptorStorePath),
```